sqlsmith: expected *DJSON, found tree.dNull

roachtest.sqlsmith/setup=seed/setting=no-mutations [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=4224429&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=4224429&tab=artifacts#/sqlsmith/setup=seed/setting=no-mutations) on master @ [83e2df701688745fe7e7d8a0d0e5e7a4ba8633c8](https://github.com/cockroachdb/cockroach/commits/83e2df701688745fe7e7d8a0d0e5e7a4ba8633c8):


```
				AS (
					SELECT
						*
					FROM
						(
							VALUES
								(
									st_box2dfromgeohash(e'\\``f\x19-W@,-':::STRING::STRING, 814595452779908034:::INT8::INT8)::BOX2D
								),
								('BOX(-0.499642664718319 -0.12018735374704459,0.6544973581088034 0.31165643492693895)':::BOX2D)
						)
							AS tab_83637 (col_205255)
				),
			with_34138 (col_205256)
				AS (
					SELECT
						*
					FROM
						(
							VALUES
								('[[[0.6545778787827349], [[], {}], [[]], false], 0.9851421085098832, [{}]]':::JSONB),
								('{"foobar": "tQBL_3x&``KI%"}':::JSONB),
								(
									e'[["foo", true], 0.9208579329159275, {"P4``M*_h": false, "b": {"baz": [], "c": {}}}, "iN``\\\\w8+", "foobar"]':::JSONB
								),
								(
									crdb_internal.merge_statement_stats((ARRAY[e'[[[]], [], {"EY1XbTeX\\\\}": [], "b": 0.4005467512334784, "y.<5y``lu": []}, {"b": 2.3487230572276037, "foobar": 3.416636623519482}, {"baz": []}]':::JSONB,e'[{"^\\\\y3R4": [[0.7805728422243667], []], "c": {"<^$": null}, "x?x@[Q{Shx*h": {"H*Mvs-4FgQ%": {}, "MK,o+-b:}0": "T PAma"}}, {"b": {"epaPI:NKG];~": {}, "gg98": {}}}, "a", 2.084382546340062]':::JSONB,'[null]':::JSONB]::JSONB[] || NULL::JSONB)::JSONB[]::JSONB[])::JSONB
								),
								('null':::JSONB),
								(
									'{"G&(r|": {"foobar": 0.2452494316955311}, "bar": {}, "baz": {"%, i>E&WoEc": []}, "c": [false]}':::JSONB
								)
						)
							AS tab_83638 (col_205256)
				)
		SELECT
			tab_83639.crdb_internal_mvcc_timestamp AS col_205257, false AS col_205258, tab_83639._date AS col_205259
		FROM
			defaultdb.public.seed@[0] AS tab_83639
		ORDER BY
			tab_83639._timestamptz ASC,
			tab_83639._int2 DESC,
			tab_83639._int4,
			tab_83639._decimal ASC,
			tab_83639._enum ASC,
			tab_83639._decimal ASC,
			tab_83639._int4 DESC,
			tab_83639._interval
		LIMIT
			4:::INT8;
```
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)

</p>
</details>
/cc @cockroachdb/sql-queries
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*sqlsmith/setup=seed/setting=no-mutations.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

```
test failure: 	sqlsmith.go:258,sqlsmith.go:297,test_runner.go:780: error: pq: internal error: expected *DJSON, found tree.dNull

CREATE TYPE greeting AS ENUM ('hello', 'howdy', 'hi', 'good day', 'morning');
CREATE TABLE IF NOT EXISTS seed AS
	SELECT
		g::INT2 AS _int2,
		g::INT4 AS _int4,
		g::INT8 AS _int8,
		g::FLOAT4 AS _float4,
		g::FLOAT8 AS _float8,
		'2001-01-01'::DATE + g AS _date,
		'2001-01-01'::TIMESTAMP + g * '1 day'::INTERVAL AS _timestamp,
		'2001-01-01'::TIMESTAMPTZ + g * '1 day'::INTERVAL AS _timestamptz,
		g * '1 day'::INTERVAL AS _interval,
		g % 2 = 1 AS _bool,
		g::DECIMAL AS _decimal,
		g::STRING AS _string,
		g::STRING::BYTES AS _bytes,
		substring('00000000-0000-0000-0000-' || g::STRING || '00000000000', 1, 36)::UUID AS _uuid,
		'0.0.0.0'::INET + g AS _inet,
		g::STRING::JSONB AS _jsonb,
		enum_range('hello'::greeting)[g] as _enum
	FROM
		generate_series(1, 5) AS g;

INSERT INTO seed DEFAULT VALUES;
CREATE INDEX on seed (_int8, _float8, _date);
CREATE INVERTED INDEX on seed (_jsonb);

		
		WITH
			with_34137 (col_205255)
				AS (
					SELECT
						*
					FROM
						(
							VALUES
								(
									st_box2dfromgeohash(e'\\`f\x19-W@,-':::STRING::STRING, 814595452779908034:::INT8::INT8)::BOX2D
								),
								('BOX(-0.499642664718319 -0.12018735374704459,0.6544973581088034 0.31165643492693895)':::BOX2D)
						)
							AS tab_83637 (col_205255)
				),
			with_34138 (col_205256)
				AS (
					SELECT
						*
					FROM
						(
							VALUES
								('[[[0.6545778787827349], [[], {}], [[]], false], 0.9851421085098832, [{}]]':::JSONB),
								('{"foobar": "tQBL_3x&`KI%"}':::JSONB),
								(
									e'[["foo", true], 0.9208579329159275, {"P4`M*_h": false, "b": {"baz": [], "c": {}}}, "iN`\\\\w8+", "foobar"]':::JSONB
								),
								(
									crdb_internal.merge_statement_stats((ARRAY[e'[[[]], [], {"EY1XbTeX\\\\}": [], "b": 0.4005467512334784, "y.<5y`lu": []}, {"b": 2.3487230572276037, "foobar": 3.416636623519482}, {"baz": []}]':::JSONB,e'[{"^\\\\y3R4": [[0.7805728422243667], []], "c": {"<^$": null}, "x?x@[Q{Shx*h": {"H*Mvs-4FgQ%": {}, "MK,o+-b:}0": "T PAma"}}, {"b": {"epaPI:NKG];~": {}, "gg98": {}}}, "a", 2.084382546340062]':::JSONB,'[null]':::JSONB]::JSONB[] || NULL::JSONB)::JSONB[]::JSONB[])::JSONB
								),
								('null':::JSONB),
								(
									'{"G&(r|": {"foobar": 0.2452494316955311}, "bar": {}, "baz": {"%, i>E&WoEc": []}, "c": [false]}':::JSONB
								)
						)
							AS tab_83638 (col_205256)
				)
		SELECT
			tab_83639.crdb_internal_mvcc_timestamp AS col_205257, false AS col_205258, tab_83639._date AS col_205259
		FROM
			defaultdb.public.seed@[0] AS tab_83639
		ORDER BY
			tab_83639._timestamptz ASC,
			tab_83639._int2 DESC,
			tab_83639._int4,
			tab_83639._decimal ASC,
			tab_83639._enum ASC,
			tab_83639._decimal ASC,
			tab_83639._int4 DESC,
			tab_83639._interval
		LIMIT
			4:::INT8;
```

--- a/pkg/sql/sem/builtins/builtins.go
+++ b/pkg/sql/sem/builtins/builtins.go
@@ -4085,6 +4085,9 @@ value if you rely on the HLC for accuracy.`,
 				arr := tree.MustBeDArray(args[0])
 				var aggregatedStats roachpb.StatementStatistics
 				for _, statsDatum := range arr.Array {
+					if statsDatum == tree.DNull {
+						continue
+					}
 					var stats roachpb.StatementStatistics
 					statsJSON := tree.MustBeDJSON(statsDatum).JSON
 					if err := sqlstatsutil.DecodeStmtStatsStatisticsJSON(statsJSON, &stats); err != nil {
@@ -4113,6 +4116,9 @@ value if you rely on the HLC for accuracy.`,
 				arr := tree.MustBeDArray(args[0])
 				var aggregatedStats roachpb.TransactionStatistics
 				for _, statsDatum := range arr.Array {
+					if statsDatum == tree.DNull {
+						continue
+					}
 					var stats roachpb.TransactionStatistics
 					statsJSON := tree.MustBeDJSON(statsDatum).JSON
 					if err := sqlstatsutil.DecodeTxnStatsStatisticsJSON(statsJSON, &stats); err != nil {

